### PR TITLE
[Caffe2] Adding pyyaml to mac and windows builds

### PR DESCRIPTION
**Pull request by pjh5**

### CODE REVIEW

**Comment by pjh5** [comment]
Cancelled all tests to just run the mac test https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-system-macos10.13-build/7412/ . None of the other tests run this code

File: scripts/build_local.sh [hunk 1]
```diff
@@ -31,6 +31,23 @@ if [ -n "${USE_ANACONDA}" ]; then
...
+      # Use pip if it's available
+      echo "Installing pyyaml with pip at $(which pip)"
+      pip install pyyaml
+    else
+      # Otherwise use the packages setup.py
+      pyyaml_dir=_temp_pyyaml_install
+      mkdir $pyyaml_dir && pushd $pyyaml_dir
+      curl http://pyyaml.org/download/pyyaml/PyYAML-3.12.tar.gz --output tarfile
+      tar -xf tarfile --strip 1
+      sudo python setup.py install
```
Comment (soumith) [inline scripts/build_local.sh]: `sudo` ??

Also, what is the case when `pip` isn't available?

Comment (pjh5) [reply scripts/build_local.sh]: We have CI builds for conda python, homebrew python, and "system" python /usr/bin that doesn't come with pip

Comment (pjh5) [reply scripts/build_local.sh]: Actually I'm going to change this to first install pip with easy_install and then use that. That's probably closer to what users will have, and if they want a useful python installation they'll probably need pip anyways.

**Comment by pjh5** [comment]
Windows first try https://ci.pytorch.org/jenkins/job/caffe2-builds/job/py2-cuda9.0-cudnn7-windows-build/4953

**Comment by pjh5** [comment]
The mac changes will fail if pip is not installed on the base machines, but that cannot be done in this script due to lack of permissions.

**Review by mingzhe09088 — APPROVED** [review]

**Pull request closed without merging**